src/Compiler/Driver/fsc.fs

```diff
@@ -689,6 +689,22 @@ let main1
     AbortOnError(diagnosticsLogger, exiter)
     ReportTime tcConfig "Typechecked"
 
+    // Fixup typars
+    let rec kindaEvilFixingEntity (entity: Entity) =
```

Comment:
I wouldn't even call it evil 😄 

Not sure if this can be solved in any manner that wouldn't involve an extra pass-through with renames.